authorization/role/repository/default_role_mapping_blackbox_test.go

```diff
@@ -0,0 +1,148 @@
+package repository_test
+
+import (
+	"testing"
+
+	rolerepo "github.com/fabric8-services/fabric8-auth/authorization/role/repository"
+	"github.com/fabric8-services/fabric8-auth/gormtestsupport"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type defaultRoleMappingBlackBoxTest struct {
+	gormtestsupport.DBTestSuite
+	repo rolerepo.DefaultRoleMappingRepository
+}
+
+func TestRunDefaultRoleMappingBlackBoxTest(t *testing.T) {
+	suite.Run(t, &defaultRoleMappingBlackBoxTest{DBTestSuite: gormtestsupport.NewDBTestSuite()})
+}
+
+func (s *defaultRoleMappingBlackBoxTest) SetupTest() {
+	s.DBTestSuite.SetupTest()
+	s.repo = rolerepo.NewDefaultRoleMappingRepository(s.DB)
+}
+
+func (s *defaultRoleMappingBlackBoxTest) TestOKToDelete() {
+	g := s.NewTestGraph()
+
+	rt := g.CreateResourceType()
+	rm := rolerepo.DefaultRoleMapping{
+		ResourceTypeID: rt.ResourceType().ResourceTypeID,
+		FromRoleID:     g.CreateRole(rt).Role().RoleID,
+		ToRoleID:       g.CreateRole().Role().RoleID,
+	}
+
+	err := s.repo.Create(s.Ctx, &rm)
+	require.NoError(s.T(), err)
+
+	rt2 := g.CreateResourceType()
+	rm2 := rolerepo.DefaultRoleMapping{
+		ResourceTypeID: rt2.ResourceType().ResourceTypeID,
+		FromRoleID:     g.CreateRole(rt).Role().RoleID,
+		ToRoleID:       g.CreateRole().Role().RoleID,
+	}
+
+	err = s.repo.Create(s.Ctx, &rm2)
+	require.NoError(s.T(), err)
+
+	mappings, err := s.repo.List(s.Ctx)
+	require.Nil(s.T(), err, "Could not list role mappings")
```

Comment:
Better to use require.NoError(). In this case the error message will also contain the info about the unexpected error.

Reply:
Fixed